qa/rpc-tests/checkblockatheight.py

```diff
@@ -35,6 +28,11 @@
 # mainchain uses this value for targeting a blockhash when building cbh script starting from chain tip backwards
 CBH_DELTA_HEIGHT = 300
 
+DISHONEST_NODE_INDEX = NUMB_OF_NODES - 1    # Dishonest node
+HONEST_NODES = list(range(NUMB_OF_NODES))
+HONEST_NODES.remove(DISHONEST_NODE_INDEX)
+MAIN_NODE = HONEST_NODES[0]
```

Comment:
I've tested changing `MAIN_NODE` to `HONEST_NODES[1]` or `HONEST_NODES[2]` and found the test is failing.
I see some options here:

- put an assert at test startup, checking `MAIN_NODE` equals `HONEST_NODES[0]` (probably the best option, zero effort)
- changing the logic/usage of `HONEST_NODES` to something like `HONEST_NODES_BUT_MAIN`
- providing a simple inline function that extract the n-th `HONEST_NODE` different from `MAIN_NODE` (and use it in the appropriate places)

Reply:
I agree: this refactor aimed at fixing the indexing of the dishonest / isolated node, but the `MAIN_NODE` alias introduces a constraint on this role. As now, I followed the first suggestion and added a note + an assert to prevent tests to fail in an unpredictable manner.